CodeEdit/Features/SourceControl/Client/GitClient+Pull.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  GitClient+Pull.swift
+//  CodeEdit
+//
+//  Created by Austin Condiff on 11/18/23.
+//
+
+import Foundation
+
+extension GitClient {
+    /// Pull changes from remote
+    func pullFromRemote(remote: String? = nil, branch: String? = nil, rebase: Bool? = nil) async throws {
+        var command = "pull"
+
+        if let remote = remote, let branch = branch {
```

Comment:
This can be done better in my opinion:

```swift
func pullFromRemote(remote: String = "origin",
                    branch: String = "main", 
                    rebase: Bool = false) async throws {
    var command = "pull"

    if remote == "" {
        print("Remote needs to be specified")
        return
    }

    if branch == "" {
        print("Branch needs to specified")
        return
    }

    command += " \(remote) \(branch)"

    if rebase {
        command += " --rebase"
    }

    _ = try await self.run(command)
}
```

Reply:
You are making the assumption that their default branch is "main".

Reply:
> You are making the assumption that their default branch is "main".

Nope, just an example, because you would always need to pull the origin and branch regardless

Reply:
I think you'd want to default to nothing, in case of remote name and branch name.

The rebase value should indeed be non-nullable. But, since there is also the option to use --merge or --ff-only. might want to make it an inner private enum, instead of a boolean.

But if you'd follow clean code, you want four functions:
`pullFromRemoteWithRebase`
`pullFromRemoteWithMerge`
`pullFromRemoteWithFastForward`
`pullFromRemoteWithGlobalConfig`
Or something like that

Reply:
I would suggest making that one function instead of multiple, you would end up with repetitive code if you make multiple functions